internal/server/smb/server.go

```diff
@@ -0,0 +1,53 @@
+package smb
+
+import (
+	"context"
+
+	"github.com/kubernetes-csi/csi-proxy/client/apiversion"
+	"github.com/kubernetes-csi/csi-proxy/internal/server/smb/internal"
+)
+
+type Server struct {
+	hostAPI API
+}
+
+type API interface {
+	NewSmbGlobalMapping(remotePath, username, password string) error
+	RemoveSmbGlobalMapping(remotePath string) error
+}
+
+func NewServer(hostAPI API) (*Server, error) {
+	return &Server{
+		hostAPI: hostAPI,
+	}, nil
+}
+
+func (s *Server) NewSmbGlobalMapping(context context.Context, request *internal.NewSmbGlobalMappingRequest, version apiversion.Version) (*internal.NewSmbGlobalMappingResponse, error) {
+	response := &internal.NewSmbGlobalMappingResponse{}
+	remotePath := request.RemotePath
+
+	if remotePath == "" {
+		response.Error = "remote path is empty"
+	}
+
+	err := s.hostAPI.NewSmbGlobalMapping(remotePath, request.Username, request.Password)
+	if err != nil {
+		response.Error = err.Error()
```

Comment:
I know we used this pattern initially in `filesystem` but this is not necessary. It's cleaner to just return the err s directly as in https://github.com/kubernetes-csi/csi-proxy/blob/master/internal/server/volume/server.go#L121, https://github.com/kubernetes-csi/csi-proxy/blob/master/internal/server/disk/server.go#L79, etc. I need to go back and fix up `filesystem` but can the SMB APIs use the more direct pattern in disk/volume API groups?

Reply:
fixed